### PR TITLE
Refactor event handlers

### PR DESCRIPTION
**Pull request by Afonso-2403**

### CODE REVIEW

File: backend/api/EventHandlers/MissionEventHandler.cs [hunk 1]
```diff
@@ -116,34 +116,21 @@ private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
...
 
-                var lastExecutedMissionRun = await MissionService.ReadLastExecutedMissionRunByRobot(robot.Id);
-                if (lastExecutedMissionRun is null)
-                {
-                    _logger.LogError("Could not find last executed mission run for robot");
-                    return;
-                }
-
-                if (!lastExecutedMissionRun.IsDriveToMission())
+                if (await ReturnToHomeService.IsRobotHome(robot.Id)) { await RobotService.UpdateCurrentArea(robot.Id, null); }
```
Comment (oysand) [inline backend/api/EventHandlers/MissionEventHandler.cs:132]: Add return here and have the functionality below withouth the else?

Comment (oysand) [reply backend/api/EventHandlers/MissionEventHandler.cs:132]: Consider removing the return on line 143 if this is done

File: backend/api/Services/ReturnToHomeService.cs [hunk 2]
```diff
@@ -4,11 +4,24 @@ namespace Api.Services
...
+        {
+            var lastExecutedMissionRun = await missionRunService.ReadLastExecutedMissionRunByRobot(robotId);
+            if (lastExecutedMissionRun is null)
+            {
+                logger.LogInformation("Could not find last executed mission run for robot {RobotId}, can not guarantee that the robot is in its home", robotId);
+                return false;
+            }
+
+            if (lastExecutedMissionRun.IsDriveToMission()) { return true; }
+            else { return false; }
```
Comment (oysand) [inline backend/api/Services/ReturnToHomeService.cs:23]:
```suggestion
            return lastExecutedMissionRun.IsDriveToMission();
```

File: backend/api/Services/RobotService.cs [hunk 3]
```diff
@@ -163,6 +165,39 @@ public async Task<IList<Robot>> ReadLocalizedRobotsForInstallation(string instal
                 .ToListAsync();
         }
 
+        public async Task OnIsarUnavailable(string robotId)
```
Comment (oysand) [inline backend/api/Services/RobotService.cs:168]: Consider moving to an event handler. Could be RobotEventHandler or something descriptive. If it is to be in the service it should be renamed so id does not sound like something that handles an event

Comment (Afonso-2403) [reply backend/api/Services/RobotService.cs:168]: Given that it is not called in the sequence of an event I kept it in `RobotService` and renamed it to `SetRobotOffline`

**Review by oysand — APPROVED** [review]
LGTM

**Pull request merged**